split lib/dma.h and start namespacing with DMA_

kv2019i opened the pull request:
Opening PR for DMA cleanup. It does seem this is going to be a bit more invasive (= need to touch a lot of code) than originally thought, so I'd like to get some feedback with this smaller PR before I go about mass renaming the whole SOF side DMA interface. 

The definitions moved out to dma-legacy.h don't need to be touched (about 300 lines worth of definitions).

The definitions that remain in zephyr/lib/dma.h however need to be prefixed, and same needs to be done in xtos/lib/dma.h as well. 

The benefit of all this is that in generic code, it's immediately clear which DMA function calls and definitions are Zephyr API calls, and which are related to the SOF DMA layer that still sits on top.

Link: https://github.com/thesofproject/sof/issues/9561

lyakh reviewed (approved):
it looks a bit confusing that the two last patches are doing the same renaming in different headers *and* C-code, but I suppose it's correct - we really do have 2 non-overlapping sets of .c and .h using the same macros which then can be renamed separately?

kv2019i commented:
@lyakh wrote:
> it looks a bit confusing that the two last patches are doing the same renaming in different headers _and_ C-code, but I suppose it's correct - we really do have 2 non-overlapping sets of .c and .h using the same macros which then can be 

Agreed, this is does look weird and I expect I'll need more tree wide big single patch for other parts of dma.h. OTOH, there is some logic to it as majority (but not all) of DMA usage is in the host/dai code and this code is also branched in calling code (e.g. there is dai-zephyr.c and dai-legacy.c). So there is cleanup work needed at multiple layers I'm afraid.

lgirdwood reviewed (approved):
LGTM @kv2019i - one step at a time.

kv2019i commented:
New version pushed:
- fixed the stub/fuzzer build fails (and cleaned up the posix DMA driver implementation)
- pivot on the approach -> not touching the XTOS drivers in this version anymore, just modify the common code and add minimal compatibility definitions to xtos/lib/dma.h (really only <10 uses in generic code so)

The main idea is to push ugly compatibility code to layers that will be removed later (with Zephyr transition) and keep the "longterm code" clean.

E.g. now it's easy to see in src/audio/host-zephyr.c that SOF_DMA_DEV_HOST is a SOF definition while DMA_ATTR_BUFFER_ADDRESS_ALIGNMENT is coming from Zephyr dma.h.

kv2019i commented:
New version:
- issue with testbench solved (compatibility also added to posix lib/dma.h

lyakh commented on zephyr/include/sof/lib/dma.h:
```diff
@@ -193,36 +177,25 @@ struct dma_chan_status {
-
-	int (*probe)(struct dma *dma);
-	int (*remove)(struct dma *dma);
+struct dma_chan_data {
```
not a reason for a new revision, but **if** you do end up doing another update - would it be possible to keep this structure at the same place as now to reduce the diff?

kv2019i replied:
@lyakh Ack, fixed in new revision.

lyakh commented on posix/include/sof/lib/dma.h:
```diff
@@ -41,6 +41,12 @@ struct comp_buffer;
+#define SOF_DMA_DIR_LMEM_TO_HMEM	DMA_DIR_LMEM_TO_HMEM
+#define SOF_DMA_DIR_MEM_TO_DEV		DMA_DIR_MEM_TO_DEV
+#define SOF_DMA_DIR_DEV_TO_MEM		DMA_DIR_DEV_TO_MEM
+#define SOF_DMA_DIR_DEV_TO_DEV		DMA_DIR_DEV_TO_DEV
```
ok, maybe the compiler doesn't care, but wouldn't it be easier for the humans to have these under right-hand definitions just below? More cases below

kv2019i replied:
@lyakh Hmm, ok, let me make a rare concession for the humans among us and change the order. Addressed in the updated version.

kv2019i replied:
Oops, sorry, I made the changes to xtos/lib/dma.h. Let me make the same to posix/lib/dma.h as well. Will make a new update in a minute.

kv2019i commented:
New version:
 - fixed build error in "instaler / checktree" CI run 
 - address Guennadi's comments (no fucntional change, reordering defintions)

kv2019i commented:
New version:
- fixed Guennadi's comment also in posix/lib/dma.h

lyakh commented:
> New version:
> 
>     * fixed Guennadi's comment also in posix/lib/dma.h

@kv2019i thanks! No more complains from me for this one so far :-)
